graph/src/data_source/common.rs

```diff
@@ -508,14 +593,22 @@ impl FromStr for CallArg {
         }
 
         let mut parts = s.split('.');
-        match (parts.next(), parts.next(), parts.next()) {
-            (Some("event"), Some("address"), None) => Ok(CallArg::Ethereum(EthereumArg::Address)),
-            (Some("event"), Some("params"), Some(param)) => {
+        match (parts.next(), parts.next(), parts.next(), parts.next()) {
+            (Some("event"), Some("address"), None, None) => {
+                Ok(CallArg::Ethereum(EthereumArg::Address))
+            }
+            (Some("event"), Some("params"), Some(param), None) => {
                 Ok(CallArg::Ethereum(EthereumArg::Param(Word::from(param))))
             }
-            (Some("entity"), Some(param), None) => Ok(CallArg::Subgraph(SubgraphArg::EntityParam(
-                Word::from(param),
-            ))),
+            (Some("event"), Some("params"), Some(param), Some(field)) if parts.next().is_none() => {
+                Ok(CallArg::Ethereum(EthereumArg::ParamField(
+                    Word::from(param),
+                    Word::from(field),
```

Comment:
If we are only allowing field indexes for struct access, we should check here already that `field` is a valid integer and store that in `ParamField`.

I am not sure how important it is to use the names of struct fields, but allowing that would require a good bit of ABI parsing so that we can turn the name of the struct field into an index

Reply:
Yeah. I think it's fair to not support named struct properties for now and just go with the numeric index. We can consider to add that later if it's requested.